File: src/WeddingBidders.WebApp/projects/wedding-bidders/src/app/core/models/wedding.ts
export interface Wedding {
  weddingId: string;
  numberOfGuests: number;
  numberOfHours: number;
  location: string;
  date: string;
  customerId?: string;
  categories: Category[];
}

export interface Category {
  categoryId: string;
  name: string;
}

export interface CreateWeddingRequest {
  numberOfGuests: number;
  numberOfHours: number;
  location: string;
  date: string;
  categories: { name: string }[];
}

export interface UpdateWeddingRequest {
  weddingId: string;
  numberOfGuests: number;
  numberOfHours: number;
  location: string;
}

File: src/WeddingBidders.Core/Model/WeddingAggregate/Wedding.cs
using WeddingBidders.Core.Model.BidAggregate;
using WeddingBidders.Core.Model.CustomerAggregate;

namespace WeddingBidders.Core.Model.WeddingAggregate;

public class Wedding
{
    public Guid WeddingId { get; set; }
    public int NumberOfGuests { get; set; }
    public int NumberOfHours { get; set; }
    public string Location { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public Guid? CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime? LastModifiedDate { get; set; }
    public ICollection<Category> Categories { get; set; } = new List<Category>();
    public ICollection<Bid> Bids { get; set; } = new List<Bid>();
}
